### PR TITLE
Fix linux build of wradlib 0.7.0

### PR DESCRIPTION
**Pull request by jjhelmus**
[skip appveyor]

### CODE REVIEW

**Comment by jjhelmus** [comment]
@pelson This fixes the Linux build issue by installing the various shared libraries in the Docker image during the build based upon the comments in the `run_docker_build.sh` scipt in the staged-recipes repo.  I'm not sure if this is the best method for doing this since all recipes which have a dependency on matplotlib would need to add this line after feedstock generation but it at least solves the immediate issue.

Also, thanks for all your work setting up conda-forge.  I've been having a great time putting these recipes together.

**Comment by ocefpaf** [comment]
@jjhelmus if you want to avoid self merging stuff you can ping me for a review and merge here.

**Comment by jjhelmus** [comment]
@ocefpaf I can't self merge at the moment. I didn't get added to the repo team for this or the other feedstocks.  I think that step may still be a manual process that @pelson or another conda-forge admin needs to trigger.  If you could review this PR and conda-forge/arm_pyart-feedstock#2 I would appreciate it.

**Comment by ocefpaf** [comment]
> @ocefpaf I can't self merge at the moment.

I add you manually.  Let me know if that works.

> I didn't get added to the repo team for this or the other feedstocks.  I think that step may still be a manual process that @pelson or another conda-forge admin needs to trigger.

As long as I know it should be automatic. (Ping @pelson.) But I can add you manually in the other repos too.

>  If you could review this PR and conda-forge/arm_pyart-feedstock#2 I would appreciate it.

Done.

**Comment by ocefpaf** [comment]
@jjhelmus I manually added you to this and other feedstocks repos that you submitted to staged-recipes.  Let me know if that worked and/or if I missed any.

**Comment by jjhelmus** [comment]
@ocefpaf Adding me to the repos worked, thanks!  I self-merged the arm_pyart pull request to test that I have merge permission and it went fine (I know bad practice, not planning on doing it again).

**Comment by ocefpaf** [comment]
> I know bad practice, not planning on doing it again.

Not sure how "bad" that will be in our context here (conda-forge and feedstocks).  We sure need a second pair of eyes when submitting to `staged-recipes` and, some big changes to the feedstocks, could also use some reviewing. However, most of the action in the feedstock will be updating the packages.  That is fine to self-merge.

**Comment by pelson** [comment]
> As long as I know it should be automatic. (Ping @pelson.) But I can add you manually in the other repos too.

It is automatic, but it only runs when the conda-forge/conda-forge.github.io travis job runs. @ocefpaf - you have the power to re-start that job on travis-ci.

**Comment by pelson** [comment]
> Not sure how "bad" that will be in our context here (conda-forge and feedstocks). We sure need a second pair of eyes when submitting to staged-recipes and, some big changes to the feedstocks, could also use some reviewing. However, most of the action in the feedstock will be updating the packages. That is fine to self-merge.

I agree. If you are a maintainer of a feedstock, you are entrusted with ensuring its best interests. If that means self merging a bumped version, I think that is OK. Ultimately, as @ocefpaf says, if you are making wider changes that have a greater uncertainty, then consulting somebody else is always a good idea.

File: ci_support/run_docker_build.sh [hunk 1]
```diff
@@ -37,6 +37,8 @@ echo "$config" > ~/.condarc
 # A lock sometimes occurs with incomplete builds. The lock file is stored in build_artefacts.
 conda clean --lock
 
+# library dependencies for matplotlib
+yum install -y libXext libXrender libSM tk libX11-devel
```
Comment (pelson) [inline ci_support/run_docker_build.sh:41]: I'm afraid these will disappear when the conda-smithy re-renders. Some of these are reasonable to expect of the OS (though I'm not sure about tk) and can be pushed up into the docker image.

Essentially, I think this is a temporary solution (which I'm OK with) that we need to find longer term solutions to.

**Comment by ocefpaf** [comment]
> It is automatic, but it only runs when the conda-forge/conda-forge.github.io travis job runs. @ocefpaf - you have the power to re-start that job on travis-ci.

Will do next time.

> Some of these are reasonable to expect of the OS.

I am having many reports on people using Docker images that lack those. (And also some OS X user!)

**Comment by jjhelmus** [comment]
I'll throw out two possible locations for the `yum install` command which would allow conda-smithy to re-render the feedstock.  The first would be to include the command in the `build.sh` script along the lines of:

``` bash
if [ `uname` == Linux ]; then
    yum install -y libXext libXrender libSM tk libX11-devel
fi
```

This has the benefit of specifying the system dependency in the recipe.  The disadvantage is that the recipe is less portable, trying to build a package from the recipe with a set up different than that used in conda-forge (e.g. a Debian based distribution) would fail. 

The other option would be have the `run_docker_build.sh` script run a recipe specific "environment setup" script when present.  Something like:

``` bash
if [ -f extra_setup.sh  ]; then
    ./extra_setup.sh
fi
```

The recipe would be more portable although the dependency would be less well specified in the recipe, although it would still be discoverable in the feedstock repo by examining the extra script.  Not sure how this extra script would be specified in the stage-recipes and transferred to the feedstock.

**Comment by jjhelmus** [comment]
> Some of these are reasonable to expect of the OS.
> I am having many reports on people using Docker images that lack those. (And also some OS X user!)

Trying to cover all possible Docker images would be hard.  The drive to make smaller and smaller Docker images means more and more libraries are being excluded from the images.  Some images are using non-glibc clibs like musl or uClibc. Getting conda packages to work on these would be very difficult.  

We might want to consider [PEP 0513's](https://www.python.org/dev/peps/pep-0513/) list of  [libraries](https://www.python.org/dev/peps/pep-0513/#the-manylinux1-policy) as being those which should be provided by the system.  The conda packages provided by Continuum seem to assume that these libraries are available.

**Comment by pelson** [comment]
> The conda packages provided by Continuum seem to assume that these libraries are available.

Actually, I thought it was the other way around - that list has been composed based on what Anaconda (and EPD) depend upon.

> The recipe would be more portable although the dependency would be less well specified in the recipe, although it would still be discoverable in the feedstock repo by examining the extra script.

Originally, I was thinking the same thing, and conda-forge.yaml has/had that space to set that up. The more I think about the infrastructure we have set up in conda-forge, the more I'm considering packaging up those libraries as conda distributions (we don't need to _build_ them, just copy them). Some headspace is needed on this one, but one example I saw in the wild is of interest: https://github.com/conda/conda-recipes/blob/79753dba37a26878811afcfe2d35d75f781db81a/pandoc/meta.yaml

**Comment by pelson** [comment]
Looks like we should be having this conversation in https://github.com/conda-forge/conda-forge.github.io/issues/23.

**Pull request merged**